pyproject.toml

```diff
@@ -94,6 +94,7 @@ crosshair-tool = ">=0.0.78"
 coverage = ">=7.6.4"
 line_profiler=">=4.2.0" #this is the minimum version which supports python 3.13
 platformdirs = ">=4.3.7"
+pre-commit = "^4.2.0"
```

Comment:
pre-commit is not a production time dependency. it will not be used by codeflash's users. it is a dev dependency